src/test/shell/bazel/toolchain_test.sh

```diff
@@ -2034,6 +2034,35 @@ EOF
   expect_not_log "does not provide ToolchainTypeInfo"
 }
 
+function test_toolchain_from_no_toolchain_rule() {
```

Comment:
What's the purpose of testing this at both the unit test and shell test levels?

Reply:
Good point, I wrote this first then added the other to debug, but we only need one. Removed the shell test.